game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1752,19 +1753,45 @@ private void resetCanDieFromReachingMaxDamage() {
   }
 
   private void setIsSuicide(final String s) {
-    isSuicide = getBool(s);
+    setIsSuicide(getBool(s));
   }
 
   private void setIsSuicide(final Boolean s) {
-    isSuicide = s;
+    setIsSuicideOnAttack(s);
+    setIsSuicideOnDefense(s && !Properties.getDefendingSuicideAndMunitionUnitsDoNotFire(getData()));
+    setIsFirstStrike(s);
```

Comment:
Should first strike truly be a side effect of is suicide? Does this limit our options in case we want a unit that is just 'isSuicide'? Say someone has a non-attacking unit, or an attacking unit that is suicide, but not first strike, how could that be handled?

Reply:
In theory no but this is for backwards compatibility because that is how suicide units work currently. Moving forward maps shouldn't use isSuicide and set either or both of the 2 new options.

Reply:
I think this is where map versions would be incredibly useful. We could decouple this based on specified map version. 

Have you considered making isSuicde have attribute values? eg: 'suicideOn=attack', 'suicideOn=all'